### ISSUE TITLE
[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.

### ISSUE THREAD

**Issue by cbchd** [thmhoag/arkserver#10]
###########################################################################

# Ark Server - Sat Jun 20 20:47:02 UTC 2020

###########################################################################

Ensuring correct permissions...

Cleaning up any leftover arkmanager files...

Creating arkmanager.cfg from environment variables...

No game files found. Installing...

Creating crontab...

Starting cron service...

* Starting periodic command scheduler cron

...done.

Loading crontab...

Save file validation is not enabled.

Backup on start is not enabled.

Running command 'start' for instance 'main'

[ WARN ] Your ARK server exec could not be found.

Checking for updates before starting

Checking for update; PID: 46

sed: can't read /ark/server/steamapps/appmanifest_376030.acf: No such file or directory

Copying to staging directory

cp: cannot stat '/ark/server/Engine/.': No such file or directory

cp: cannot stat '/ark/server/linux64/.': No such file or directory

cp: cannot stat '/ark/server/ShooterGame/Content/Mods/111111111/.': No such file or directory

cp: cannot stat '/ark/server/ShooterGame/Content/Mods/111111111.mod': No such file or directory

cp: cannot stat '/ark/server/steamapps/.': No such file or directory

Downloading ARK updateExecuting /home/steam/.steam/steamcmd/steamcmd.sh +@NoPromptForPassword 1 +login anonymous +force_install_dir /ark/staging +app_update 376030 +quit

Redirecting stderr to '/home/steam/.steam/logs/stderr.txt'

[ 0%] Checking for available updates...

[----] Verifying installation...

Steam Console Client (c) Valve Corporation

-- type 'quit' to exit --

Loading Steam API...Warning: failed to init SDL thread priority manager: SDL not found

OK.

"@NoPromptForPassword" = "1"

Connecting anonymously to Steam Public...Logged in OK

Waiting for user info...OK

Update state (0x3) reconfiguring, progress: 0.00 (0 / 0)

Update state (0x3) reconfiguring, progress: 0.00 (0 / 0)

Update state (0x5) verifying install, progress: 7.93 (1015738408 / 12808440321)

Update state (0x5) verifying install, progress: 14.61 (1871294658 / 12808440321)

Update state (0x5) verifying install, progress: 82.90 (10618201471 / 12808440321)

Update state (0x5) verifying install, progress: 86.58 (11089972111 / 12808440321)

Update state (0x11) preallocating, progress: 8.26 (1058239362 / 12808440321)

Update state (0x11) preallocating, progress: 9.50 (1217413491 / 12808440321)

Update state (0x11) preallocating, progress: 11.31 (1448258352 / 12808440321)

Update state (0x11) preallocating, progress: 12.96 (1659824536 / 12808440321)

Update state (0x11) preallocating, progress: 14.86 (1902703156 / 12808440321)

Update state (0x11) preallocating, progress: 15.16 (1941555744 / 12808440321)

Update state (0x11) preallocating, progress: 49.97 (6400893701 / 12808440321)

Update state (0x11) preallocating, progress: 65.80 (8427386961 / 12808440321)

Update state (0x11) preallocating, progress: 83.36 (10676617049 / 12808440321)

Update state (0x11) preallocating, progress: 84.19 (10783429998 / 12808440321)

Update state (0x11) preallocating, progress: 85.14 (10904723714 / 12808440321)

Update state (0x11) preallocating, progress: 86.25 (11046999495 / 12808440321)

Update state (0x11) preallocating, progress: 87.03 (11146632955 / 12808440321)

Update state (0x11) preallocating, progress: 87.77 (11241954163 / 12808440321)

Update state (0x11) preallocating, progress: 99.69 (12768226904 / 12808440321)

Update state (0x61) downloading, progress: 0.18 (22875584 / 12808440321)

Update state (0x61) downloading, progress: 0.93 (118610302 / 12808440321)

Update state (0x61) downloading, progress: 1.97 (252533350 / 12808440321)

Update state (0x61) downloading, progress: 3.04 (389736638 / 12808440321)

Update state (0x61) downloading, progress: 3.78 (483645544 / 12808440321)

Update state (0x61) downloading, progress: 4.65 (596164799 / 12808440321)

Update state (0x61) downloading, progress: 5.74 (735841915 / 12808440321)

Update state (0x61) downloading, progress: 6.75 (864934936 / 12808440321)

Update state (0x61) downloading, progress: 7.45 (954356463 / 12808440321)

Update state (0x61) downloading, progress: 8.37 (1072215590 / 12808440321)

Update state (0x61) downloading, progress: 8.88 (1137204258 / 12808440321)

Update state (0x61) downloading, progress: 9.02 (1155458819 / 12808440321)

Update state (0x61) downloading, progress: 9.16 (1173521704 / 12808440321)

Update state (0x61) downloading, progress: 9.27 (1186974341 / 12808440321)

Update state (0x61) downloading, progress: 9.42 (1206873561 / 12808440321)

Update state (0x61) downloading, progress: 9.55 (1223581854 / 12808440321)

Update state (0x61) downloading, progress: 9.74 (1247986347 / 12808440321)

Update state (0x61) downloading, progress: 9.92 (1270719115 / 12808440321)

Update state (0x61) downloading, progress: 10.12 (1295618208 / 12808440321)

Update state (0x61) downloading, progress: 10.27 (1314998699 / 12808440321)

Update state (0x61) downloading, progress: 10.54 (1350063594 / 12808440321)

Update state (0x61) downloading, progress: 10.76 (1377990880 / 12808440321)

Update state (0x61) downloading, progress: 10.90 (1396142006 / 12808440321)

Update state (0x61) downloading, progress: 11.09 (1421005346 / 12808440321)

Update state (0x61) downloading, progress: 11.19 (1432921137 / 12808440321)

Update state (0x61) downloading, progress: 11.46 (1467826830 / 12808440321)

Update state (0x61) downloading, progress: 11.63 (1489493898 / 12808440321)

Update state (0x61) downloading, progress: 11.85 (1517316191 / 12808440321)

Update state (0x61) downloading, progress: 12.04 (1542562925 / 12808440321)

Update state (0x61) downloading, progress: 12.19 (1561637784 / 12808440321)

Update state (0x61) downloading, progress: 12.37 (1584337811 / 12808440321)

Update state (0x61) downloading, progress: 12.54 (1606008145 / 12808440321)

Update state (0x61) downloading, progress: 12.84 (1645171955 / 12808440321)

Update state (0x61) downloading, progress: 13.09 (1676286842 / 12808440321)

Update state (0x61) downloading, progress: 13.16 (1685667804 / 12808440321)

Update state (0x61) downloading, progress: 13.36 (1711169440 / 12808440321)

Update state (0x61) downloading, progress: 13.53 (1732652796 / 12808440321)

Update state (0x61) downloading, progress: 13.57 (1738170457 / 12808440321)

Update state (0x61) downloading, progress: 13.70 (1754551585 / 12808440321)

Update state (0x61) downloading, progress: 13.83 (1772044284 / 12808440321)

Update state (0x61) downloading, progress: 13.88 (1778428045 / 12808440321)

Update state (0x61) downloading, progress: 13.98 (1791030747 / 12808440321)

Update state (0x61) downloading, progress: 14.08 (1803245463 / 12808440321)

Update state (0x61) downloading, progress: 14.13 (1809413095 / 12808440321)

Update state (0x61) downloading, progress: 14.20 (1818789827 / 12808440321)

Update state (0x61) downloading, progress: 15.33 (1963022396 / 12808440321)

Update state (0x61) downloading, progress: 16.08 (2059901695 / 12808440321)

Update state (0x61) downloading, progress: 16.53 (2117625192 / 12808440321)

Update state (0x61) downloading, progress: 17.00 (2177857469 / 12808440321)

Update state (0x61) downloading, progress: 17.38 (2225484690 / 12808440321)

Update state (0x61) downloading, progress: 17.86 (2286980857 / 12808440321)

Update state (0x61) downloading, progress: 18.36 (2351165144 / 12808440321)

Update state (0x61) downloading, progress: 18.88 (2417910582 / 12808440321)

Update state (0x61) downloading, progress: 19.38 (2482125464 / 12808440321)

Update state (0x61) downloading, progress: 19.96 (2556803677 / 12808440321)

Update state (0x61) downloading, progress: 20.72 (2653995381 / 12808440321)

Update state (0x61) downloading, progress: 21.26 (2723074162 / 12808440321)

Update state (0x61) downloading, progress: 21.76 (2787310420 / 12808440321)

Update state (0x61) downloading, progress: 22.47 (2877751701 / 12808440321)

Update state (0x61) downloading, progress: 22.90 (2933167271 / 12808440321)

Update state (0x61) downloading, progress: 23.29 (2983017423 / 12808440321)

Update state (0x61) downloading, progress: 23.83 (3052213367 / 12808440321)

Update state (0x61) downloading, progress: 24.42 (3127991445 / 12808440321)

Update state (0x61) downloading, progress: 25.16 (3222350001 / 12808440321)

Update state (0x61) downloading, progress: 25.63 (3283287896 / 12808440321)

Update state (0x61) downloading, progress: 26.24 (3361032178 / 12808440321)

Update state (0x61) downloading, progress: 26.67 (3415841052 / 12808440321)

Update state (0x61) downloading, progress: 27.16 (3479176418 / 12808440321)

Update state (0x61) downloading, progress: 28.00 (3586905412 / 12808440321)

Update state (0x61) downloading, progress: 28.73 (3680402914 / 12808440321)

Update state (0x61) downloading, progress: 29.32 (3755358829 / 12808440321)

Update state (0x61) downloading, progress: 29.78 (3814901165 / 12808440321)

Update state (0x61) downloading, progress: 30.27 (3877591673 / 12808440321)

Update state (0x61) downloading, progress: 30.72 (3935029061 / 12808440321)

Update state (0x61) downloading, progress: 31.25 (4003130533 / 12808440321)

Update state (0x61) downloading, progress: 31.83 (4076304332 / 12808440321)

Update state (0x61) downloading, progress: 32.44 (4154864180 / 12808440321)

Update state (0x61) downloading, progress: 33.01 (4228510747 / 12808440321)

Update state (0x61) downloading, progress: 33.57 (4299852927 / 12808440321)

Update state (0x61) downloading, progress: 34.18 (4377708968 / 12808440321)

Update state (0x61) downloading, progress: 34.96 (4477929491 / 12808440321)

Update state (0x61) downloading, progress: 35.97 (4606959722 / 12808440321)

Update state (0x61) downloading, progress: 36.72 (4702894549 / 12808440321)

Update state (0x61) downloading, progress: 37.29 (4775679226 / 12808440321)

Update state (0x61) downloading, progress: 37.84 (4847021436 / 12808440321)

Update state (0x61) downloading, progress: 38.34 (4910639164 / 12808440321)

Update state (0x61) downloading, progress: 38.81 (4971164826 / 12808440321)

Update state (0x61) downloading, progress: 39.36 (5040822196 / 12808440321)

Update state (0x61) downloading, progress: 39.90 (5110111772 / 12808440321)

Update state (0x61) downloading, progress: 40.50 (5187688912 / 12808440321)

Update state (0x61) downloading, progress: 41.10 (5263899368 / 12808440321)

Update state (0x61) downloading, progress: 41.74 (5345965734 / 12808440321)

Update state (0x61) downloading, progress: 42.33 (5421597185 / 12808440321)

Update state (0x61) downloading, progress: 42.80 (5482317309 / 12808440321)

Update state (0x61) downloading, progress: 43.42 (5561782456 / 12808440321)

Update state (0x61) downloading, progress: 43.85 (5616226872 / 12808440321)

Update state (0x61) downloading, progress: 44.28 (5671427018 / 12808440321)

Update state (0x61) downloading, progress: 44.73 (5729081343 / 12808440321)

Update state (0x61) downloading, progress: 45.17 (5785655603 / 12808440321)

Update state (0x61) downloading, progress: 45.74 (5858049652 / 12808440321)

Update state (0x61) downloading, progress: 46.25 (5923963244 / 12808440321)

Update state (0x61) downloading, progress: 46.75 (5988163015 / 12808440321)

Update state (0x61) downloading, progress: 47.25 (6052176229 / 12808440321)

Update state (0x61) downloading, progress: 47.96 (6142848942 / 12808440321)

Update state (0x61) downloading, progress: 48.59 (6224187408 / 12808440321)

Update state (0x61) downloading, progress: 49.21 (6302433568 / 12808440321)

Update state (0x61) downloading, progress: 49.77 (6375165506 / 12808440321)

Update state (0x61) downloading, progress: 50.19 (6428617031 / 12808440321)

Update state (0x61) downloading, progress: 50.79 (6506012896 / 12808440321)

Update state (0x61) downloading, progress: 51.34 (6575453575 / 12808440321)

Update state (0x61) downloading, progress: 52.09 (6672379043 / 12808440321)

Update state (0x61) downloading, progress: 52.68 (6748093127 / 12808440321)

Update state (0x61) downloading, progress: 53.20 (6814455429 / 12808440321)

Update state (0x61) downloading, progress: 53.85 (6896953154 / 12808440321)

Update state (0x61) downloading, progress: 54.36 (6962133094 / 12808440321)

Update state (0x61) downloading, progress: 54.96 (7040123362 / 12808440321)

Update state (0x61) downloading, progress: 55.75 (7140204815 / 12808440321)

Update state (0x61) downloading, progress: 56.30 (7211297621 / 12808440321)

Update state (0x61) downloading, progress: 56.93 (7291704356 / 12808440321)

Update state (0x61) downloading, progress: 57.40 (7352505858 / 12808440321)

Update state (0x61) downloading, progress: 57.93 (7420408028 / 12808440321)

Update state (0x61) downloading, progress: 58.48 (7490417342 / 12808440321)

Update state (0x61) downloading, progress: 59.02 (7559127011 / 12808440321)

Update state (0x61) downloading, progress: 59.57 (7630006274 / 12808440321)

Update state (0x61) downloading, progress: 60.13 (7701318406 / 12808440321)

Update state (0x61) downloading, progress: 60.68 (7771869463 / 12808440321)

Update state (0x61) downloading, progress: 61.27 (7847866267 / 12808440321)

Update state (0x61) downloading, progress: 61.80 (7915702822 / 12808440321)

Update state (0x61) downloading, progress: 62.33 (7983050553 / 12808440321)

Update state (0x61) downloading, progress: 62.91 (8057354026 / 12808440321)

Update state (0x61) downloading, progress: 63.43 (8124273940 / 12808440321)

Update state (0x61) downloading, progress: 63.96 (8191640645 / 12808440321)

Update state (0x61) downloading, progress: 64.44 (8253219538 / 12808440321)

Update state (0x61) downloading, progress: 64.95 (8319142339 / 12808440321)

Update state (0x61) downloading, progress: 65.46 (8384701514 / 12808440321)

Update state (0x61) downloading, progress: 65.98 (8451470381 / 12808440321)

Update state (0x61) downloading, progress: 66.51 (8518590459 / 12808440321)

Update state (0x61) downloading, progress: 67.08 (8592061809 / 12808440321)

Update state (0x61) downloading, progress: 67.56 (8653395952 / 12808440321)

Update state (0x61) downloading, progress: 68.10 (8722726570 / 12808440321)

Update state (0x61) downloading, progress: 68.70 (8799005991 / 12808440321)

Update state (0x61) downloading, progress: 69.04 (8842350989 / 12808440321)

Update state (0x61) downloading, progress: 69.51 (8903364416 / 12808440321)

Update state (0x61) downloading, progress: 70.19 (8989968162 / 12808440321)

Update state (0x61) downloading, progress: 70.97 (9090739291 / 12808440321)

Update state (0x61) downloading, progress: 71.59 (9169424942 / 12808440321)

Update state (0x61) downloading, progress: 72.15 (9241254324 / 12808440321)

Update state (0x61) downloading, progress: 72.77 (9320251939 / 12808440321)

Update state (0x61) downloading, progress: 73.34 (9393585946 / 12808440321)

Update state (0x61) downloading, progress: 73.82 (9455718042 / 12808440321)

Update state (0x61) downloading, progress: 74.33 (9521044494 / 12808440321)

Update state (0x61) downloading, progress: 74.90 (9593200595 / 12808440321)

Update state (0x61) downloading, progress: 75.38 (9654558729 / 12808440321)

Update state (0x61) downloading, progress: 76.01 (9735549979 / 12808440321)

Update state (0x61) downloading, progress: 76.51 (9799657871 / 12808440321)

Update state (0x61) downloading, progress: 76.97 (9859116161 / 12808440321)

Update state (0x61) downloading, progress: 77.54 (9931060044 / 12808440321)

Update state (0x61) downloading, progress: 78.20 (10015899521 / 12808440321)

Update state (0x61) downloading, progress: 78.79 (10091459177 / 12808440321)

Update state (0x61) downloading, progress: 79.35 (10163393707 / 12808440321)

Update state (0x61) downloading, progress: 79.84 (10226663450 / 12808440321)

Update state (0x61) downloading, progress: 80.41 (10299348004 / 12808440321)

Update state (0x61) downloading, progress: 80.89 (10360634366 / 12808440321)

Update state (0x61) downloading, progress: 81.33 (10417378719 / 12808440321)

Update state (0x61) downloading, progress: 81.80 (10477841968 / 12808440321)

Update state (0x61) downloading, progress: 82.28 (10539342318 / 12808440321)

Update state (0x61) downloading, progress: 82.82 (10608324807 / 12808440321)

Update state (0x61) downloading, progress: 83.22 (10659478091 / 12808440321)

Update state (0x61) downloading, progress: 83.72 (10723482444 / 12808440321)

Update state (0x61) downloading, progress: 84.25 (10790553368 / 12808440321)

Update state (0x61) downloading, progress: 84.81 (10863101811 / 12808440321)

Update state (0x61) downloading, progress: 85.26 (10920331405 / 12808440321)

Update state (0x61) downloading, progress: 85.79 (10988104238 / 12808440321)

Update state (0x61) downloading, progress: 86.28 (11051659875 / 12808440321)

Update state (0x61) downloading, progress: 86.74 (11109681911 / 12808440321)

Update state (0x61) downloading, progress: 87.16 (11163808803 / 12808440321)

Update state (0x61) downloading, progress: 87.61 (11221002939 / 12808440321)

Update state (0x61) downloading, progress: 88.09 (11283283858 / 12808440321)

Update state (0x61) downloading, progress: 88.54 (11340459744 / 12808440321)

Update state (0x61) downloading, progress: 89.01 (11400679645 / 12808440321)

Update state (0x61) downloading, progress: 89.50 (11463397888 / 12808440321)

Update state (0x61) downloading, progress: 90.13 (11544280031 / 12808440321)

Update state (0x61) downloading, progress: 90.79 (11629081767 / 12808440321)

Update state (0x61) downloading, progress: 91.31 (11695104160 / 12808440321)

Update state (0x61) downloading, progress: 91.84 (11762766610 / 12808440321)

Update state (0x61) downloading, progress: 92.44 (11840237840 / 12808440321)

Update state (0x61) downloading, progress: 93.13 (11928997761 / 12808440321)

Update state (0x61) downloading, progress: 93.75 (12008334414 / 12808440321)

Update state (0x61) downloading, progress: 94.31 (12079380770 / 12808440321)

Update state (0x61) downloading, progress: 94.92 (12157463112 / 12808440321)

Update state (0x61) downloading, progress: 95.44 (12223765255 / 12808440321)

Update state (0x61) downloading, progress: 95.98 (12293259760 / 12808440321)

Update state (0x61) downloading, progress: 96.45 (12353177002 / 12808440321)

Update state (0x61) downloading, progress: 96.93 (12415564296 / 12808440321)

Update state (0x61) downloading, progress: 97.39 (12474402225 / 12808440321)

Update state (0x61) downloading, progress: 97.88 (12537428914 / 12808440321)

Update state (0x61) downloading, progress: 98.36 (12598159362 / 12808440321)

Update state (0x61) downloading, progress: 98.87 (12663211775 / 12808440321)

Update state (0x61) downloading, progress: 99.40 (12731809467 / 12808440321)

Update state (0x61) downloading, progress: 99.89 (12794111147 / 12808440321)

Update state (0x101) committing, progress: 11.06 (1416372285 / 12808440321)

Update state (0x101) committing, progress: 15.12 (1936022763 / 12808440321)

Update state (0x101) committing, progress: 54.58 (6990808541 / 12808440321)

Update state (0x101) committing, progress: 83.60 (10707702455 / 12808440321)

Update state (0x101) committing, progress: 86.48 (11076261856 / 12808440321)

Update state (0x101) committing, progress: 99.69 (12769325672 / 12808440321)

Success! App '376030' fully installed.

CWorkThreadPool::~CWorkThreadPool: work processing queue not empty: 1 items discarded.

The server is already stopped

Applying update from staging directory

Update to 5192336 complete

The server is starting...

2020-06-20 20:55:10: start

2020-06-20 20:55:10: Running /ark/server/ShooterGame/Binaries/Linux/ShooterGameServer TheIsland\?RCONPort=32330\?SessionName=Ark\ Server\?MaxPlayers=70\?QueryPort=27015\?ServerAdminPassword=k3yb04rdc4t\?Port=7777\?RCONEnabled=True\?ServerPassword\?listen -log

2020-06-20 20:55:10: Server PID: 413

[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.

Setting breakpad minidump AppID = 346110

/usr/local/bin/arkmanager: line 911: 413 Killed "$arkserverroot/$arkserverexec" "$arkserveropts" "${arkextraopts[@]}"

2020-06-20 21:05:41: Bad PID ''; expected '413'

2020-06-20 21:05:41: exited with status 0

###########################################################################

# Ark Server - Sun Jun 21 13:35:14 UTC 2020

###########################################################################

Ensuring correct permissions...

Cleaning up any leftover arkmanager files...

Creating arkmanager.cfg from environment variables...

Starting cron service...

* Starting periodic command scheduler cron

...done.

Loading crontab...

Save file validation is not enabled.

"Backup on start is not enabled.

Running command 'start' for instance 'main'

Checking for updates before starting

Checking for update; PID: 41

Your server is already up to date! The most recent version is 5192336.

The server is starting...

2020-06-21 13:35:24: start

2020-06-21 13:35:24: Running /ark/server/ShooterGame/Binaries/Linux/ShooterGameServer TheIsland\?RCONPort=32330\?SessionName=Ark\ Server\?MaxPlayers=70\?QueryPort=27015\?ServerAdminPassword=k3yb04rdc4t\?Port=7777\?RCONEnabled=True\?ServerPassword\?listen -log

2020-06-21 13:35:24: Server PID: 128

[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed."

## Image Setup Details

* **Image Tag**: 
OS: Windows 10
* **Docker Host OS**: 
Docker Desktop

Docker command: 
docker run -d -v steam:/home/steam/Steam -v ark:/ark -p 27015:27015 -p 27015:27015/udp -p 7778:7778 -p 7778:7778/udp  -p 7777:7777 -p 7777:7777/udp thmhoag/arkserver

Description of issue
Server is not starting when used with the docker command. Getting an error about steam not being running - "[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed."

**Comment by thmhoag** [comment]
Hey @cbchd, have you tried connecting to the server even with this error? This error is common with the linux version of the server and is kind of a red herring. All of my servers see this error on every startup but continue to function without issue.

**Comment by cbchd** [comment]
Thanks thmhoag for getting back to me! I'll try it out and get back to you in a few hours :-D

**Comment by cbchd** [comment]
Server boots up and I can connect on steam, yay! One of the issues we had was not allocating 6 GB of RAM to the docker container, so the server never fully started up and was hitting the m.2 drive at 100% for 25mins.

We tried getting epic games to work with the server, but currently all we get is a timeout error, with port 7777 port-forwarded, crossplay enabled, and battleye disabled, while trying to connect with singleplayer console command to open the connection. But that's a separate issue. Thanks for the help!

**Comment by thmhoag** [comment]
No problem @cbchd, glad you were able to figure it out. 👍  I'll go ahead and close the issue.
